app/controllers/concerns/json_collection_pagination.rb

```diff
@@ -4,56 +4,47 @@ module JsonCollectionPagination
   SERIALIZER = nil
 
   def paginated_json_collection(collection, path_helper, sort_key, sort_order, offset, per_page, total, params)
-    #changing from order to reorder to discard previous ordering
-    sort_key = (sort_key.presence || :id).to_sym
-    sort_order = sort_order.to_s == 'desc' ? :desc : :asc
-    fail ArgumentError.new('Invalid sort_key') unless collection.column_names.include?(sort_key.to_s)
-    collection = collection.reorder(sort_key => sort_order)
+    # Backwards compatibility
+    params[:sort_key] = sort_key
+    params[:sort_order] = sort_order
+    params[:offset] = offset
+    params[:per_page] = per_page
+    params[:total] = total
+    self.params = params
+    paginated_json_collection_new(collection)
+  end
 
+  def paginated_json_collection_new(collection)
     # Meta
-    offset = (offset.presence || 0).to_i
-    include_total = (total == true || total == 'true')
+    per_page = sort_per_page
+    offset = sort_offset
+    include_total = sort_total
     meta = {
-        sort_key: sort_key,
-        sort_order: sort_order,
-        offset: offset
+      sort_key: sort_key,
+      sort_order: sort_order,
+      offset: offset,
+      per_page: per_page
     }
 
-    if [false, 'false', '∞'].include?(per_page)
-      # allow pagination to be disabled
-      # worst case: the query takes longer than 2 minutes and is
-      # killed at the database and the load balancer
-      meta[:per_page] = per_page
-      data_on_page = collection
-    else
-      per_page = (per_page.presence || self.class::PER_PAGE).to_i
-      meta[:per_page] = per_page
-      # Get the current page of results.
-      #  Add +1 to limit to see if there is a next page.
-      #  This will be dropped in the return.
```

Comment:
I think this comment is worth retaining to explain why there's that `per_page+1`